types/node/module.d.ts

```diff
@@ -685,6 +685,30 @@ declare module "module" {
              * @returns The absolute URL string that the specifier would resolve to.
              */
             resolve(specifier: string, parent?: string | URL): string;
+            /**
+             * `true` when the current module is the entry point of the current process; `false` otherwise.
+             *
+             * Equivalent to `require.main === module` in CommonJS.
+             *
+             * Analogous to Python's `__name__ == "__main__"`.
+             *
+             * ```js
+             * export function foo() {
+             *   return 'Hello, world';
+             * }
+             *
+             * function main() {
+             *   const message = foo();
+             *   console.log(message);
+             * }
+             *
+             * if (import.meta.main) main();
+             * // `foo` can be imported from another module without possible side-effects from `main`
+             * ```
+             * @since v24.2.0
```

Comment:
The `import.meta.main` was introduced in v24.2.0, but it there is a [bug when using with TypeScript](https://github.com/nodejs/node/issues/58660) that was fixed in v24.3.0.

I suggest adding a comment:

```diff
+  * > **Caveat**: broken in v24.2.x when entry point is a TypeScript file
    * @since v24.2.0
```